app/src/main/java/io/horizontalsystems/bankwallet/modules/send/SendActivity.kt

```diff
@@ -95,44 +95,25 @@ class SendActivity : BaseActivity() {
             when (input) {
                 SendModule.Input.Amount -> {
                     //add amount view
-                    val amountViewModel = ViewModelProviders.of(this).get(SendAmountViewModel::class.java)
-                    val amountPresenter = amountViewModel.init(wallet, mainViewModel.amountModuleDelegate)
-
-                    mainPresenter.handler.amountModule = amountPresenter
-
-                    val amountView = SendAmountView(context = this, lifecycleOwner = this, sendAmountViewModel = amountViewModel)
-                    sendLinearLayout.addView(amountView)
-                    amountView.requestFocus()
+                    val sendAmountFragment = SendAmountFragment(wallet, mainPresenter)
```

Comment:
It is better to pass only required items to SendAmountFragment constructor. SendAmountFragment shouldn't know about SendPresenter. So let's pass ISendHandler and IAmountModuleDelegate